### PR TITLE
[READY] tested code

### PR DESCRIPTION
**Pull request by lolman360**
i forgot how byond if() works for agood second there

this was tested in coderbus's bot channel and i'm compiling it rn

### CODE REVIEW

**Comment by lolman360** [comment]
Verified ✔️

File: code/modules/projectiles/guns/ballistic.dm [hunk 1]
```diff
@@ -53,6 +53,10 @@
 	..()
 	if (istype(A, /obj/item/ammo_box/magazine))
 		var/obj/item/ammo_box/magazine/AM = A
+		if(AM.load_delay)
+			if(do_after(user, AM.load_delay, target = src))
+			else
+				return FALSE
```
Comment (silicons) [inline code/modules/projectiles/guns/ballistic.dm]:
```suggestion
		if(AM.load_delay && !do_after(user, AM.load_delay, target = src))
			return FALSE
```

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm [hunk 2]
```diff
@@ -76,21 +76,28 @@
 	return 1
 
 /obj/item/ammo_box/attackby(obj/item/A, mob/user, params, silent = FALSE, replace_spent = 0)
-	if(INTERACTING_WITH(user, A))
+	if(INTERACTING_WITH(user, src) || INTERACTING_WITH(user, A))
```
Comment (silicons) [inline code/modules/projectiles/boxes_magazines/_box_magazine.dm:79]:
```suggestion
	if(INTERACTING_WITH(user, src))
```

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm [hunk 3]
```diff
@@ -76,21 +76,28 @@
...
-					num_loaded++
-				if(!did_load || !multiload)
-					break
+			if(load_delay || AM.load_delay)
+				var/loadtime = load_delay
+				if(AM.load_delay > load_delay)
+					loadtime = AM.load_delay
+				if(do_after(user, loadtime, target = src))
+				else
+					return FALSE
```
Comment (silicons) [inline code/modules/projectiles/boxes_magazines/_box_magazine.dm]:
```suggestion
				if(!do_after(user, loadtime, target = src))
					return FALSE
```

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm [hunk 4]
```diff
@@ -76,21 +76,28 @@
...
-				var/did_load = give_round(AC, replace_spent)
-				if(did_load)
-					AM.stored_ammo -= AC
-					num_loaded++
-				if(!did_load || !multiload)
-					break
+			if(load_delay || AM.load_delay)
+				var/loadtime = load_delay
+				if(AM.load_delay > load_delay)
+					loadtime = AM.load_delay
```
Comment (silicons) [inline code/modules/projectiles/boxes_magazines/_box_magazine.dm]:
```suggestion
			if(load_delay || AM.load_delay)
				var/loadtime = max(AM.load_delay, load_delay)
```

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm (hunk 2 above)
Comment (lolman360) [reply code/modules/projectiles/boxes_magazines/_box_magazine.dm:79]: i tested it without both and half the time the check didn't work so

**Pull request merged**